Update to sbt 1.2.8 (adding --ignore-source-errors for javadoc)

raboof opened the pull request:
`--ignore-source-errors` makes 'sbt -Dakka.genjavadoc.enabled=true clean unidoc' ignore source errors.

The genjavadoc-generated code in `akka.stream.scaladsl.GraphDSL.Implicits.ReversePortsOps` in `scaladsl/GraphDSL.java` seems to be problematic, but since this is scaladsl anyway it seems reasonable to just exclude it from the javadoc.

Refs #26057, #26100

raboof committed:
Update to 1.2.8 and add --ignore-source-errors for javadoc

`--ignore-source-errors` makes 'sbt -Dakka.genjavadoc.enabled=true clean unidoc'
at least output something to target/javaunidoc, though AFAICS it is not
complete (missing index, CSS etc)

raboof commented:
Next step is probably to run the build with a debugger attached, putting a breakpoint at `jdk.javadoc.internal.tool.Main#execute` and rummaging about how the error handling there works.

helena commented:
@raboof want to assign #26100 to you since these are marked related now and this resolves that?

helena commented on project/Doc.scala:
```diff
@@ -115,7 +115,7 @@ object UnidocRoot extends AutoPlugin {
 
   val akkaSettings = UnidocRoot.CliOptions.genjavadocEnabled.ifTrue(
-    Seq(javacOptions in (JavaUnidoc, unidoc) := Seq("-Xdoclint:none"))).getOrElse(Nil)
+    Seq(javacOptions in (JavaUnidoc, unidoc) := Seq("-Xdoclint:none", "--ignore-source-errors"))).getOrElse(Nil)
```
Nice, is `--ignore-source-errors` still undocumented?

raboof replied:
AFAICT yes

helena reviewed (approved):
LGTM - the failing two tests are finicky, I've seen them fail and then pass. We need to look at that.

helena commented:
PLS BUILD ALL

helena commented:
PLS BUILD

raboof commented:
@helena I'm happy to take on #26100 further (after next week), but I'm afraid this WIP PR is not quite ready to be merged yet: even with `--ignore-source-errors`, `sbt -Dakka.genjavadoc.enabled=true clean unidoc` does not yet produce a full javadoc HTML hierarchy on this branch yet (while it does on master)

raboof committed:
Skip javadoc for scaladsl/GraphDSL.java

As problematic javadoc seems to be generated for
akka.stream.scaladsl.GraphDSL.Implicits.ReversePortsOps

raboof committed:
Enable frames again

raboof commented:
Updated: I think this should do it :crossed_fingers:

patriknw reviewed (approved):
LGTM

cunei commented:
Note: the flag `--ignore-source-errors` is not recognized by JDK 1.8.0_202, which means doc generation fails in that case.

raboof commented:
@cunei you're right! This doesn't seem like a big deal since we require jdk9+ for releasing anyway, but might be good to mention in CONTRIBUTING.md. Added #26317
